test/integration/swim-test.js

```diff
@@ -0,0 +1,198 @@
+            'number of selected ping-req members is correct');
+        assert.equal(unreachableMember.status, 'alive',
+            'unreachable member is alive');
+        assert.end();
+    });
+});
+
+testRingpopCluster({
+    size: 5,
+    tap: function tap(cluster) {
+        mkNoGossip(cluster);
+    }
+}, 'ping-req target unreachable', function t(bootRes, cluster, assert) {
+    var badRingpop = cluster[4];
+    badRingpop.destroy();
+
+    var ringpop = cluster[0];
+    var unreachableMember = ringpop.membership.findMemberByAddress(badRingpop.hostPort);
+    var pingReqSize = 3;
+
+    sendPingReq({
+        ringpop: ringpop,
+        unreachableMember: unreachableMember,
+        pingReqSize: pingReqSize
+    }, function onPingReq(err, res) {
+        assert.ifErr(err, 'no error occurred');
+        assertNumBadStatuses(assert, res, pingReqSize);
+        assert.equals(unreachableMember.status, 'suspect',
+            'unreachable member is suspect');
+        assert.end();
+    });
+});
+
+testRingpopCluster({
+    size: 2,
+    tap: function tap(cluster) {
+        mkNoGossip(cluster);
+    }
+}, 'no ping-req members', function t(bootRes, cluster, assert) {
+    var ringpop = cluster[0];
+    var ringpop2Addr = cluster[1].hostPort;
+
+    var unreachableMember = ringpop.membership.findMemberByAddress(ringpop2Addr);
+    var pingReqSize = 3;
+
+    sendPingReq({
+        ringpop: ringpop,
+        unreachableMember: unreachableMember,
+        pingReqSize: pingReqSize
+    }, function onPingReq(err, res) {
+        assert.ok(err, 'error occurred');
+        assert.equals(err.type, 'ringpop.ping-req.no-members', 'No members error');
+        assert.end();
+    });
+});
+
+testRingpopCluster({
+    size: 5,
+    tap: function tap(cluster) {
+        mkBadPingReqResponder(cluster[3]);
+        mkNoGossip(cluster);
+    }
+}, 'some bad ping-statuses', function t(bootRes, cluster, assert) {
+    var badRingpop = cluster[4];
+    badRingpop.destroy();
+
+    var ringpop = cluster[0];
+    var unreachableMember = ringpop.membership.findMemberByAddress(badRingpop.hostPort);
+    var pingReqSize = 3;
+
+    sendPingReq({
+        ringpop: ringpop,
+        unreachableMember: unreachableMember,
+        pingReqSize: pingReqSize
+    }, function onPingReq(err, res) {
+        assert.ifErr(err, 'no error occurred');
+        assertNumBadStatuses(assert, res, pingReqSize - 1);
+        assert.equals(unreachableMember.status, 'suspect',
+            'unreachable member is suspect');
+        assert.end();
+    });
+});
+
+testRingpopCluster({
+    size: 5,
+    tap: function tap(cluster) {
+        mkNoGossip(cluster);
+    }
+}, 'ping-req inconclusive', function t(bootRes, cluster, assert) {
+    var ringpop = cluster[0];
+    var unreachableMember = ringpop.membership.findMemberByAddress(cluster[4].hostPort);
+    var pingReqSize = 3;
+
+    ringpop.membership.members.forEach(function eachMember(member) {
```

Comment:
consider leaving a comment here about "mutating address of peers so that the outgoing pings to peers fails".